sound/soc/sof/topology.c

```diff
@@ -183,6 +183,23 @@ static int set_up_volume_table(struct snd_sof_control *scontrol,
 	return 0;
 }
 
+/* Can this pipeline run when driver is D3 ? */
+int snd_sof_pipeline_permit_d0i3(struct snd_sof_dev *sdev,
+				 int pipeline_id)
+{
+	struct snd_sof_widget *swidget = NULL;
+
+	list_for_each_entry(swidget, &sdev->widget_list, list) {
+		if (swidget->pipeline_id != pipeline_id)
+			continue;
+		if (swidget->pm_permit_d0i3)
+			return 0;
+	}
+
+	return 0;
```

Comment:
@lgirdwood this  seems incorrect. Should we return 0 even if the pipeline is not found?